credentials/apps/badges/models.py

```diff
@@ -231,6 +235,10 @@ class Meta:
 
     def __str__(self):
         return f"BadgeProgress:{self.username}"
+    
+    @staticmethod
+    def reset(username: str):
```

Comment:
This must be a specific user progress reset - not generic.
```
filter(progress=self)
```
So only related to the progress record fulfillments are removed.